This patch can be removed once Python is rebuilt for cygwin-1.7 (getaddrinfo).

--- origsrc/python-mpd-0.2.1/mpd.py	2009-03-08 13:59:13.265625000 -0500
+++ src/python-mpd-0.2.1/mpd.py	2009-03-08 13:58:03.250000000 -0500
@@ -290,17 +290,12 @@ class MPDClient(object):
     def connect(self, host, port):
         if self._sock:
             raise ConnectionError("Already connected")
-        msg = "getaddrinfo returns an empty list"
-        try:
-            flags = socket.AI_ADDRCONFIG
-        except AttributeError:
-            flags = 0
-        for res in socket.getaddrinfo(host, port, socket.AF_UNSPEC,
-                                      socket.SOCK_STREAM, socket.IPPROTO_TCP,
-                                      flags):
-            af, socktype, proto, canonname, sa = res
+        msg = "gethostbyname_ex returned an empty list"
+        hostname, aliaslist, ipaddrlist = socket.gethostbyname_ex(host)
+        for res in ipaddrlist:
+            sa = (res, port)
             try:
-                self._sock = socket.socket(af, socktype, proto)
+                self._sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_TCP)
                 self._sock.connect(sa)
             except socket.error, msg:
                 if self._sock:
